test/resources/data_race_different_keys/expected_output.xml

```diff
@@ -0,0 +1,80 @@
+<?xml version="1.0" ?>
+<testsuite tests="1">
+        <testcase errors="2" name="test_communication">
+                <error count="1" location="eprosima::fastrtps::rtps::UDPTransportInterface::CloseInputChannel(eprosima::fastrtps::rtps::Locator_t const&amp;) /ros2_ws/src/eProsima/Fast-RTPS/src/cpp/transp
+ort/UDPTransportInterface.cpp:108 (libfastrtps.so.1+0xX)" type="data_race">    #X asio::detail::reactive_socket_service_base::construct(asio::detail::reactive_socket_service_base::base_implementation_type
+&amp;) /usr/include/asio/detail/impl/reactive_socket_service_base.ipp:44 (libfastrtps.so.1+0xX)
+    #X asio::detail::reactive_socket_service_base::close(asio::detail::reactive_socket_service_base::base_implementation_type&amp;, std::error_code&amp;) /usr/include/asio/detail/impl/reactive_socket_serv
```

Comment:
I'll rebase on master and make changes as required.

Reply:
The `0xX` replacement happens only in stack trace `key` and not the actual `lines`.

Reply:
The actual output won't look like this then? I didn't think we did any replacement at all for the sample stack trace.

Reply:
Ah, I see. We're not actually checking that the stack trace output matches anything, which is probably right since it should be effectively random which stack trace ends up being the sample.

If what I said isn't wrong, I'd suggest getting rid of all this expected output and make something more like what we did for the expected csv "sample_stack_trace" field output.

Reply:
That is correct. We're checking that it has non-zero length and starts with a `#`. 

Sure, I can replace it with a fixed string like we did for the csv output.